packages/vanilla/src/components/horizontalrule/horizontalrule.template.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Horizontal Rule template literal
+ *
+ * @param {object} props Property object passed into the template
+ * @param {string} props.style solid vs dashed [''|'dashed']
+ * @param {string} props.size size of the hr [''|'small'|'medium'|'large']
+ * @param {string} props.contrast color of the rule [''|'low'|'high']
+ * @param {string} props.weight weight of the rule [''|'thick']
+ * @returns {string} ES6 template literal of the horizontal rule
+ */
+const horizontalruleTemplate = ({ style, size, contrast, weight }) => {
```

Comment:
@jeffchew I think this is a cool addition, but the part I'm struggling with a little in this package is this is less about the `HTML`, and more about the `Javascript`, right? It's hard with the given example `<HorizontalRule />` since it's doesn't have any given Javascript. How do you see components like `Masthead` or `Footer` fitting in with this direction since they'll need to be instantiated?

```html
<footer class="bx--footer">
  ...
</footer>
```

```javascript
dotcom.footer.init({ type: 'short' );
```

Do we add it to the page, assume it's been added via whatever framework before initializing

Reply:
Since they will ultimately be ES6 classes, they would need to get instantiated. I was digging into the carbon code, and at least for the "kitchen sink" artifact that is generated, there is a flag called "autoinit" which a user can have all of the loaded classes autoinit on page load or not. For our repo, I turned that into an environment variable and called it `VANILLA_AUTOINIT`. 

To this point, it may be worth adding this to the package level README, but at the same time, Carbon team seemed to have purposely left this flag out from their documentation. It's on by default, so that means it would have to be manually set to turn off this behavior.

As for if a user is pulling in these classes adhoc (which will be most cases), it's up to them to instantiate once their application is ready, and for all intents and purposes, they might prefer this. We'd just need to document in the README for the corresponding components that have Javascript functionality to include the init function and how they can trigger it.

cc: @annawen1 @kennylam @sannminwin @andysherman2121

Reply:
I don't think we want people using the kitchen sink approach, or do we? Just for additional context within the discussion Carbon has adopters initialize it this way...

```scss
@import '~carbon-components/scss/components/card/card';
```
```javascript
import { Card } from 'carbon-components';

const cardElement = document.getElementById('card');
const cardInstance = Card.create(cardElement);
```
https://www.carbondesignsystem.com/get-started/develop/vanilla#javascript

Reply:
We definitely don't want to encourage the kitchen sink approach. The README for the components will need to document how to instantiate the JS for that specific component.